### ISSUE TITLE
TreeChecker warnings while porting to 2.11

### ISSUE THREAD

**Issue by VladUreche** [miniboxing/miniboxing-plugin#104]
Collections did not generate out-of-scope symbols after `minibox-spec` but we got 4 new issues:

```
Compiling gh-bug-82-2.scala                                            ... [FAIL]
Diff: 
--- output
+++ expected
@@ -2,2 +2,3 @@
 [Not checkable: minibox-adapt]
 [Now checking: minibox-spec]
+warning: Reference to uninitialized variable t

Compiling gh-bug-82-3.scala                                            ... [FAIL]
Diff: 
--- output
+++ expected
@@ -2,2 +2,3 @@
 [Not checkable: minibox-adapt]
 [Now checking: minibox-spec]
+warning: Reference to uninitialized variable t

Compiling gh-bug-83-2.scala                                            ... [FAIL]
Diff: 
--- output
+++ expected
@@ -6,2 +6,3 @@
 [Not checkable: minibox-adapt]
 [Now checking: minibox-spec]
+warning: Reference to uninitialized variable t

Compiling gh-bug-94.scala                                              ... [FAIL]
Diff: 
--- output
+++ expected
@@ -1,3 +1,5 @@
 [Now checking: minibox-dupl]
+[check: minibox-dupl] symbol out of scope
+[check: minibox-dupl] symbol out of scope
 [Not checkable: minibox-adapt]
 [Now checking: minibox-spec]
```

**Comment by VladUreche** [comment]
Neither of the errors seem legitimate to me, but I'll have to dig deeper to understand why the error is generated.

**Comment by VladUreche** [comment]
Note to self: After fixing these, remove the line with "Reference to uninitialized ..." from TestSuite.scala.

**Comment by VladUreche** [comment]
`Reference to uninitialized variable x` warnings were fixed by #126. The only problem left is symbol out of scope, which I think is caused by type parameters leaking outside their scopes. TODO: Check.
